### ISSUE TITLE
Hazelcast Simulator. Cannot connect to Agents.

### ISSUE THREAD

**Issue by hamanhlam** [hazelcast/hazelcast#5560]
Environment: Windows 7. Hazelcast simulator 0.5. Using Cygwin sshd to remote agents.
Hi.
    I am trying to install hazelcast simulator following this article  http://docs.hazelcast.org/docs/3.4/manual/html/installingsimulator.html.

At last step i ran this command to try connecting my agent instance in localhost:
$ ssh -o BatchMode=yes gitpm@10.8.15.126 "echo ok" 2>&1
output:
Permission denied (publickey,password,keyboard-interactive).

After that, I try this command:
```$ ssh -v  gitpm@10.8.15.126```
output:
OpenSSH_6.8p1, OpenSSL 1.0.2c 12 Jun 2015
debug1: Reading configuration data /etc/ssh_config
debug1: Connecting to 10.8.15.126 [10.8.15.126] port 22.
debug1: Connection established.
debug1: identity file /home/gitpm/.ssh/id_rsa type 1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_rsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_dsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_dsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_ecdsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_ecdsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_ed25519 type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/gitpm/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.8
debug1: Remote protocol version 2.0, remote software version OpenSSH_6.8
debug1: match: OpenSSH_6.8 pat OpenSSH* compat 0x04000000
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: server->client aes128-ctr umac-64-etm@openssh.com none
debug1: kex: client->server aes128-ctr umac-64-etm@openssh.com none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:xMwO83PwDmOKGui/Q5oUQqa/ofyqSN0PNODCHcuZXnw
debug1: Host '10.8.15.126' is known and matches the ECDSA host key.
debug1: Found key in /home/gitpm/.ssh/known_hosts:1
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: Roaming not allowed by server
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /home/gitpm/.ssh/id_rsa
debug1: Server accepts key: pkalg ssh-rsa blen 279
Enter passphrase for key '/home/gitpm/.ssh/id_rsa':
debug1: Authentication succeeded (publickey).
Authenticated to 10.8.15.126 ([10.8.15.126]:22).
debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
Last login: Tue Jun 23 09:23:40 2015 from 10.8.15.126

And run test:
--------------------------------------------------------
gitpm@gitpm /cygdrive/d/development/hazelcast-simulator-0.5/my-tests
```$ ./run.sh```

output:
D:\development\hazelcast-simulator-0.5
INFO  11:15:19 Hazelcast Simulator Coordinator
INFO  11:15:19 Version: 0.5, Commit: c6e82c5, Build Time: 14.05.2015 @ 13:46:17 CEST
INFO  11:15:19 SIMULATOR_HOME: D:\development\hazelcast-simulator-0.5
INFO  11:15:19 Loading simulator.properties: D:\development\hazelcast-simulator-0.5\my-tests\simulator.properties
INFO  11:15:19 Loading testsuite file: D:\development\hazelcast-simulator-0.5\my-tests\test.properties
INFO  11:15:19 Loading Hazelcast configuration: D:\development\hazelcast-simulator-0.5\my-tests\..\conf\hazelcast.xml
INFO  11:15:19 Loading Hazelcast client configuration: D:\development\hazelcast-simulator-0.5\my-tests\..\conf\client-hazelcast.xml
INFO  11:15:19 Loading Log4j configuration for worker: D:\development\hazelcast-simulator-0.5\conf\worker-log4j.xml
INFO  11:15:19 Loading agents file: D:\development\hazelcast-simulator-0.5\my-tests\agents.txt
INFO  11:15:19 HAZELCAST_VERSION_SPEC: outofthebox
INFO  11:15:19 --------------------------------------------------------------
INFO  11:15:19 Waiting for agents to start
INFO  11:15:19 --------------------------------------------------------------
WARN  11:15:40 Failed to connect to public address: 10.8.15.126 sleeping for 1 second and trying again

Something wrong? Help me plz. 
Thanks.

**Comment by pveentjer** [comment]
Can you do a ssh -v simulator@10.8.15.126 and post the output.

And where are your target machines running? 10.8.15.126 sounds like an ec2 ip address.

**Comment by hamanhlam** [comment]
Thank you.
I have changed some of the information on simulator config:
GROUP_NAME=group_test
USER=gitpm

10.8.15.126 that is my ip not ec2 ip address. 
I want to test in local so I created SSH server on my machine, and  setup public/private key pair  to simulator look like a local machine remote another machine. In 10.8.15.126 (my machine) I have cluster with  2 members 10.8.15.126:5702 and 10.8.15.126:5703.
